dev/io.openliberty.netty.internal/src/io/openliberty/netty/internal/NettyFramework.java

```diff
@@ -0,0 +1,298 @@
+        serverChannels = new DefaultChannelGroup(GlobalEventExecutor.INSTANCE);
+//      udpGroup = new NioEventLoopGroup(0, executorService);
+
+        /*
+         * As a demo, this starts up a HTTP server at localhost:9081 which will return 
+         * an 200 OK response to any request
+         */
+        // use the executor service provided by Liberty
+        parentGroup = new NioEventLoopGroup(0, executorService);
+        workerGroup = new NioEventLoopGroup(0, executorService);
+
+        serverBootstrap = createTCPBoostrap(new HttpServerInitializer(), parentGroup, workerGroup);
+        try {
+            runWhenServerStarted(new Callable<Void>() {
+                @Override
+                public Void call() {
+                    try {
+                        finishInitServerSocket(serverBootstrap, 9081);
+                    } catch (Exception x) {
+                        // TODO Auto-generated catch block
+                        x.printStackTrace();
+                    }
+                    return null;
+                }
+            });
+        } catch (Exception x) {
+            // TODO Auto-generated catch block
+            x.printStackTrace();
+        }
+
+    }
+
+    @Deactivate
+    protected void deactivate(Map<String, Object> properties, int reason) {
+        shutdown();
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        shutdown();
+        activate(config);
+    }
+    
+    /**
+     * DS method for setting the executor service reference.
+     *
+     * @param executorService the {@link java.util.concurrent.ExecutorService} to
+     *                            queue work to.
+     */
+    @Reference(service = ExecutorService.class,
+               cardinality = ReferenceCardinality.MANDATORY)
+    protected void setExecutorService(ExecutorService executorService) {
+        this.executorService = executorService;
+    }
+
+    /**
+     * DS method for clearing the executor service reference.
+     * This is a required reference, will be called after deactivate.
+     *
+     * @param executorService the service instance to clear
+     */
+    protected void unsetExecutorService(ExecutorService executorService) {
+    }
+
+    /**
+     * Creates a UDP Bootstrap with the given ChannelInitializer
+     * @return Bootstrap
+     */
+    private Bootstrap createUDPBootstrap(ChannelInitializer<Channel> initializer) {
+        return new Bootstrap()
+                .group(udpGroup)
+                .channel(NioDatagramChannel.class)
+                .handler(initializer);
+    }
+
+    /**
+     * Creates a NIO TCP ServerBootstrap with the given ChannelInitializer
+     * @return ServerBootstrap
+     */
+    private ServerBootstrap createTCPBoostrap(ChannelInitializer<Channel> initializer, EventLoopGroup parentGroup, 
+            EventLoopGroup workerGroup) {
+        return new ServerBootstrap()
+                .group(this.parentGroup, this.workerGroup)
```

Comment:
It is odd that we get the terms parent for an acceptor/server and child for a client. In Vert.x terms, worker is something that is not on the channel's eventloop but initiated async from a separate worker pool. All fine for this purpose and I guess I will get more used to it.